deepchem/models/keras_model.py

```diff
@@ -957,6 +961,33 @@ def get_global_step(self):
     """Get the number of steps of fitting that have been performed."""
     return int(self._global_step)
 
+  def compute_loss(self, dataset, transformers=[]):
+    """Computes the data-fit loss term on the given dataset.
+
+    Parameters
+    ----------
+    dataset: dc.data.Dataset,
+        Dataset to compute loss on
+    transformers: list of dc.trans.Transformers
+      Transformers that the dataset has been transformed by. The output
+      is passed through these transformers to undo the transformations.
+    """
+    if hasattr(self, 'mode'):
+      if self.mode == 'regression':
+        y_true = undo_transforms(dataset.y, transformers=transformers)
+      else:
+        y_true = to_one_hot(dataset.y.flatten(), self.n_classes).reshape(
+            -1, len(self.n_tasks), self.n_classes)
```

Comment:
This is incorrect.  We shouldn't be hardcoding transforms like this.  The logic for computing the loss should exactly match what's done in `fit_generator()`.

Reply:
I was confused about this: whether evaluate the loss on the transformed output, or the base one. I decided to stick the base output, but I will change this.